user_tools/tests/spark_rapids_tools_e2e/features/steps/e2e_utils.py

```diff
@@ -0,0 +1,224 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+This module defines utility functions used by the end-to-end tests using behave.
+"""
+
+import logging
+import os
+import subprocess
+from attr import dataclass
+from enum import auto
+from pathlib import Path
+from typing import List
+from urllib.parse import urlparse
+
+from spark_rapids_tools import EnumeratedType
+
+
+@dataclass
+class E2ETestUtils:
+
+    @staticmethod
+    def get_cmd_output_str(cmd_result: subprocess.CompletedProcess) -> str:
+        """
+        Get the output of a command as a string (stdout and stderr).
+        """
+        dash_line = '-' * 50
+        cmd_sections = [("COMMAND", ' '.join(cmd_result.args)),
+                        ("STDOUT", cmd_result.stdout),
+                        ("STDERR", cmd_result.stderr)]
+        output_sections = []
+        for label, content in cmd_sections:
+            content = content.strip() if content else "No output"
+            output_sections.append(f"{dash_line}\n{label}\n{dash_line}\n{content}\n{dash_line}")
+        return '\n'.join(output_sections).strip() + '\n'
+
+    @classmethod
+    def run_sys_cmd(cls, cmd: list) -> subprocess.CompletedProcess:
+        """
+        Run a system command and return the result.
+        If verbose mode is enabled by the behave config, print the command and its output
+        """
+        cmd_result = subprocess.run(cmd, capture_output=True, text=True)
+        if cls.is_verbose_mode():
+            print(cls.get_cmd_output_str(cmd_result))
+        return cmd_result
+
+    @classmethod
+    def assert_sys_cmd_return_code(cls,
+                                   cmd_result: subprocess.CompletedProcess,
+                                   exp_return_code: int = 0,
+                                   error_msg: str = None) -> None:
+        assert cmd_result.returncode == exp_return_code, \
+            f"{error_msg}\n{cls.get_cmd_output_str(cmd_result)}"
+
+    @classmethod
+    def create_spark_rapids_cmd(cls,
+                                event_logs: List[str],
+                                output_dir: str,
+                                platform: str = 'onprem',
+                                filter_apps: str = 'all') -> List[str]:
+        """
+        Create the command to run the Spark Rapids qualification tool.
+        """
+        return [
+            cls.get_spark_rapids_cli(),
+            'qualification',
+            '--platform', platform,
+            '--eventlogs', ','.join(event_logs),
+            '-o', output_dir,
+            '--tools_jar', cls.get_tools_jar_file(),
+            '--verbose',
+            '--filter_apps', filter_apps
```

Comment:
in the future might be nice to have a way to add to these or have tests pass more specific things through, not needed now

Reply:
That makes sense. Added an TODO for this.